### PR TITLE
Add HTTP.Headers to the examples and change provider Alamofire provid…

### LINKED ISSUE #51: Update Readme
We need to update the README to reflect all the changes that we did when removing `Alamofire`

### PR DESCRIPTION
**Pull request by syllerim**
Closes #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,14 +40,18 @@ struct Post: JSONObjectInitializable {
...
+    } 
+
     func allPosts(for date: Date) -> Observable<[Post]> {
         return requestObservable(at: .posts(for: date))
     }
 }
 
 //Convenient typealias to save space :)
-typealias Connection = ApiConnection<Alamofire.DataRequest>
+typealias Connection = ApiConnection<URLSessionDataTask.DataRequest>
```
Comment (alickbass) [inline README.md]: This is wrong. Should be `ApiConnection<URLSessionDataTask>`

File: README.md [hunk 2]
```diff
@@ -64,9 +68,10 @@ If you don't really want to create your own `APIConnectionType` class, you can u
 **Example of getting posts with `NetworkConnector`:**
 
 ```swift
-typealias Connection = NetworkConnector<Alamofire.DataRequest, Api>
+typealias Connection = NetworkConnector<URLSessionDataTask.DataRequest, Api>
+var headers : HTTP.Headers = [.accept: "application/json", .contentType: "charset=utf-8", .acceptLanguage: "en-US"]
```
Comment (alickbass) [inline README.md]: This is not necessary here. You can leave it as it was

File: README.md [hunk 3]
```diff
@@ -64,9 +68,10 @@ If you don't really want to create your own `APIConnectionType` class, you can u
 **Example of getting posts with `NetworkConnector`:**
 
 ```swift
-typealias Connection = NetworkConnector<Alamofire.DataRequest, Api>
+typealias Connection = NetworkConnector<URLSessionDataTask.DataRequest, Api>
```
Comment (alickbass) [inline README.md]: Should be `NetworkConnector<URLSessionDataTask, Api>`

File: README.md [hunk 4]
```diff
@@ -64,9 +68,10 @@ If you don't really want to create your own `APIConnectionType` class, you can u
...
+var headers : HTTP.Headers = [.accept: "application/json", .contentType: "charset=utf-8", .acceptLanguage: "en-US"]
 
-let posts: Observable<[Post]> = Connection().requestObservable(at: .posts(for: Date()))
+let posts: Observable<[Post]> = Connection().requestObservable(at: .posts(for: Date()), headers: headers)
```
Comment (alickbass) [inline README.md]: You can leave it as it was before

File: README.md [hunk 5]
```diff
@@ -77,7 +82,7 @@ let disposable = posts.subscribe(onNext: { posts in
 ## Mocking Requests
 The most powerfull feature of this framework is that it is really easy to mock requests for the purpose of `Unit Testing` without running any external processes and changing any code base.
 
-The reason why `ApiConnection<Provider: DataRequestType>` is generic is to allow plugging in mock `Connection Provider`. As we saw above, for the real requests we would use `Alamofire.DataRequest` as a `Provider` for `ApiConnection`. Alamofire would make the real requests to the real servers.
+The reason why `ApiConnection<Provider: DataRequestType>` is generic is to allow plugging in mock `Connection Provider`. As we saw above, for the real requests we would use `URLSessionDataTask.DataRequest` as a `Provider` for `ApiConnection`. URLSessionDataTask would make the real requests to the real servers.
```
Comment (alickbass) [inline README.md]: should be `URLSessionDataTask` instead of `URLSessionDataTask.DataRequest`

**Review by alickbass — CHANGES_REQUESTED** [review]

**Review by alickbass — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (syllerim) [reply README.md]: 👍🏿

File: README.md (hunk 4 above)
Comment (syllerim) [reply README.md]: 👍🏿

File: README.md (hunk 2 above)
Comment (syllerim) [reply README.md]: 👍🏿

File: README.md (hunk 3 above)
Comment (syllerim) [reply README.md]: 👍🏿

File: README.md (hunk 1 above)
Comment (syllerim) [reply README.md]: 👍🏿

**Pull request merged**